control/control_test.go

```diff
@@ -136,127 +136,133 @@ func getPort() int {
-				})
-				Convey("So swapped plugins event should show plugin type as collector", func() {
-					So(lpe.plugin.PluginType, ShouldEqual, int(plugin.CollectorPluginType))
-				})
 			})
-		})
 
-		// Swap plugin with a different type of plugin
-		Convey("First plugin in catalog", t, func() {
-			Convey("Should have name mock", func() {
-				So(c.PluginCatalog()[0].Name(), ShouldEqual, "mock")
+			mockRP, mErr := core.NewRequestedPlugin(fixtures.PluginPathMock1)
+			Convey("Requested collector plugin should not error", func() {
+				So(mErr, ShouldBeNil)
 			})
-		})
 
-		filePath := strings.Replace(fixtures.PluginPathMock2, "snap-plugin-collector-mock2", "snap-plugin-publisher-mock-file", 1)
-		fileRP, pErr := core.NewRequestedPlugin(filePath)
-		Convey("Loading a plugin should not error", t, func() {
-			So(pErr, ShouldBeNil)
-		})
-		err = c.SwapPlugins(fileRP, c.PluginCatalog()[0])
-		Convey("Swapping mock and file plugins", t, func() {
-			Convey("Should error", func() {
-				So(err, ShouldNotBeNil)
+			err = c.SwapPlugins(mockRP, c.PluginCatalog()[0])
+			Convey("Swapping plugins with a different version", func() {
+				Convey("Should not error", func() {
+					So(err, ShouldBeNil)
+				})
 			})
-		})
 
-		//
-		// TODO: Write a proper rollback test as previous test was not testing rollback
-		//
-
-		// Rollback will throw an error if a plugin can not unload
+			if err != nil {
+				t.FailNow()
+			}
+			<-lpe.done
+
+			// Swap plugin that was loaded with a different version of the plugin
+			Convey("Successfull swapping plugins", func() {
+				Convey("Should generate a swapped plugins event", func() {
+					Convey("So first plugin in catalog after swap should have name mock", func() {
+						So(c.PluginCatalog()[0].Name(), ShouldEqual, "mock")
+					})
+					Convey("So swapped plugins event should show loaded plugin name as mock", func() {
+						So(lpe.plugin.LoadedPluginName, ShouldEqual, "mock")
+					})
+					Convey("So swapped plugins event should show loaded plugin version as 1", func() {
+						So(lpe.plugin.LoadedPluginVersion, ShouldEqual, 1)
+					})
+					Convey("So swapped plugins event should show unloaded plugin name as mock", func() {
+						So(lpe.plugin.UnloadedPluginName, ShouldEqual, "mock")
+					})
+					Convey("So swapped plugins event should show unloaded plugin version as 2", func() {
+						So(lpe.plugin.UnloadedPluginVersion, ShouldEqual, 2)
+					})
+					Convey("So swapped plugins event should show plugin type as collector", func() {
+						So(lpe.plugin.PluginType, ShouldEqual, int(plugin.CollectorPluginType))
+					})
+				})
 
-		Convey("Rollback failure returns error", t, func() {
-			lp := c.PluginCatalog()[0]
-			pm := new(MockPluginManagerBadSwap)
-			pm.ExistingPlugin = lp
-			c.pluginManager = pm
+			})
+			Convey("Swap plugin with a different type of plugin", func() {
+				filePath := strings.Replace(fixtures.PluginPathMock2, "snap-plugin-collector-mock2", "snap-plugin-publisher-mock-file", 1)
```

Comment:
Consider adding path to fixtures or just use `github.com/intelsdi-x/snap/plugin/helper.helper.PluginFilePath`.